### ISSUE TITLE
Function rule operates on procedure and vice versa

### ISSUE THREAD

**Issue by imd1** [jeremiah-c-leary/vhdl-style-guide#639]
**Environment**

VSG 3.3.0, Centos 7.6.1810

**Describe the bug**

Similar function and procedure rules (e.g. procedure_401 and function_012) are triggered on the "wrong" type of subprogram, leading to two sets of rules operating on the same code.  For example, rule procedure_401 operates on a function (and procedure) and function_012 operates on a procedure (and function)

In check only mode, you get two violations instead of 1 (minor problem)
In fix mode, if the rules are configured differently (would they ever be?), you get unfixed errors where 1 rule is applied and the other rule clashes

Need to review all procedure and function rule for the scope of this problem

**Comment by imd1** [comment]
A question about depreciated rules...if I reference a depreciated rule in a configuration file, what will happen?

**Comment by jeremiah-c-leary** [comment]
VSG will report an error and stop running on that file.

For example, if you try this configuration:
```yaml
  1
  2 rule:
  3   procedure_201:
  4     disabled : False
```
and run it on a file you will get:
```mono
$vsg -c deprecated_rule.yaml -f fifo.vhd -p1
ERROR: Invalid configuration of file original.vhd
ERROR [config-001] Rule procedure_201 has been depricated.
  Rule procedure_201 has been moved to rule subprogram_body_201.
```
You need the -p1 because there is still an issue with multiprocessing I need to track down.  Otherwise VSG will hang waiting for the other threads to complete.

VSG reports which rule you are attempting to configure has been deprecated and gives some information about it.  In this case rule `procedure_201` has been moved to rule `subprogram_body_201`.

The intent is to notify the user that something has changed and how to remedy it.

**Comment by jeremiah-c-leary** [comment]
I have merged this to master and will close this issue.
